{
	"id": "ALSA-2022:6595",
	"title": "Moderate: nodejs and nodejs-nodemon security and bug fix update",
	"description": "Node.js is a software development platform for building fast and scalable network applications in the JavaScript programming language. \n\nThe following packages have been upgraded to a later upstream version: nodejs (16.16.0), nodejs-nodemon (2.0.19). (BZ#2124230, BZ#2124233)\n\nSecurity Fix(es):\n\n* nodejs-ini: Prototype pollution via malicious INI file (CVE-2020-7788)\n* nodejs-glob-parent: Regular expression denial of service (CVE-2020-28469)\n* nodejs-ansi-regex: Regular expression denial of service (ReDoS) matching ANSI escape codes (CVE-2021-3807)\n* normalize-url: ReDoS for data URLs (CVE-2021-33502)\n* nodejs: npm pack ignores root-level .gitignore and .npmignore file exclusion directives when run in a workspace (CVE-2022-29244)\n* nodejs: DNS rebinding in --inspect via invalid IP addresses (CVE-2022-32212)\n* nodejs: HTTP request smuggling due to flawed parsing of Transfer-Encoding (CVE-2022-32213)\n* nodejs: HTTP request smuggling due to improper delimiting of header fields (CVE-2022-32214)\n* nodejs: HTTP request smuggling due to incorrect parsing of multi-line Transfer-Encoding (CVE-2022-32215)\n* got: missing verification of requested URLs allows redirects to UNIX sockets (CVE-2022-33987)\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.\n\nBug Fix(es):\n\n* nodejs:16/nodejs: Rebase to the latest Nodejs 16 release [almalinux-9] (BZ#2121019)\n* nodejs: Specify --with-default-icu-data-dir when using bootstrap build (BZ#2124299)",
	"affected": [
		{
			"vulnerable": true,
			"package": {
				"ecosystem": "AlmaLinux:9",
				"name": "nodejs"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "1:16.16.0-1.el9_0"
						}
					]
				}
			]
		},
		{
			"vulnerable": true,
			"package": {
				"ecosystem": "AlmaLinux:9",
				"name": "nodejs-docs"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "1:16.16.0-1.el9_0"
						}
					]
				}
			]
		},
		{
			"vulnerable": true,
			"package": {
				"ecosystem": "AlmaLinux:9",
				"name": "nodejs-full-i18n"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "1:16.16.0-1.el9_0"
						}
					]
				}
			]
		},
		{
			"vulnerable": true,
			"package": {
				"ecosystem": "AlmaLinux:9",
				"name": "nodejs-libs"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "1:16.16.0-1.el9_0"
						}
					]
				}
			]
		},
		{
			"vulnerable": true,
			"package": {
				"ecosystem": "AlmaLinux:9",
				"name": "nodejs-nodemon"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "2.0.19-1.el9_0"
						}
					]
				}
			]
		},
		{
			"vulnerable": true,
			"package": {
				"ecosystem": "AlmaLinux:9",
				"name": "npm"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "1:8.11.0-1.16.16.0.1.el9_0"
						}
					]
				}
			]
		}
	],
	"vulnerabilities": [
		{
			"cve": "CVE-2022-32213",
			"references": [
				{
					"name": "CVE-2022-32213",
					"source": "errata.almalinux.org",
					"tags": [
						"cve"
					],
					"url": "https://access.redhat.com/security/cve/CVE-2022-32213"
				}
			]
		},
		{
			"cve": "CVE-2022-32215",
			"references": [
				{
					"name": "CVE-2022-32215",
					"source": "errata.almalinux.org",
					"tags": [
						"cve"
					],
					"url": "https://access.redhat.com/security/cve/CVE-2022-32215"
				}
			]
		},
		{
			"cve": "CVE-2020-28469",
			"references": [
				{
					"name": "CVE-2020-28469",
					"source": "errata.almalinux.org",
					"tags": [
						"cve"
					],
					"url": "https://access.redhat.com/security/cve/CVE-2020-28469"
				}
			]
		},
		{
			"cve": "CVE-2021-33502",
			"references": [
				{
					"name": "CVE-2021-33502",
					"source": "errata.almalinux.org",
					"tags": [
						"cve"
					],
					"url": "https://access.redhat.com/security/cve/CVE-2021-33502"
				}
			]
		},
		{
			"cve": "CVE-2022-29244",
			"references": [
				{
					"name": "CVE-2022-29244",
					"source": "errata.almalinux.org",
					"tags": [
						"cve"
					],
					"url": "https://access.redhat.com/security/cve/CVE-2022-29244"
				}
			]
		},
		{
			"cve": "CVE-2022-32212",
			"references": [
				{
					"name": "CVE-2022-32212",
					"source": "errata.almalinux.org",
					"tags": [
						"cve"
					],
					"url": "https://access.redhat.com/security/cve/CVE-2022-32212"
				}
			]
		},
		{
			"cve": "CVE-2020-7788",
			"references": [
				{
					"name": "CVE-2020-7788",
					"source": "errata.almalinux.org",
					"tags": [
						"cve"
					],
					"url": "https://access.redhat.com/security/cve/CVE-2020-7788"
				}
			]
		},
		{
			"cve": "CVE-2021-3807",
			"references": [
				{
					"name": "CVE-2021-3807",
					"source": "errata.almalinux.org",
					"tags": [
						"cve"
					],
					"url": "https://access.redhat.com/security/cve/CVE-2021-3807"
				}
			]
		},
		{
			"cve": "CVE-2022-32214",
			"references": [
				{
					"name": "CVE-2022-32214",
					"source": "errata.almalinux.org",
					"tags": [
						"cve"
					],
					"url": "https://access.redhat.com/security/cve/CVE-2022-32214"
				}
			]
		},
		{
			"cve": "CVE-2022-33987",
			"references": [
				{
					"name": "CVE-2022-33987",
					"source": "errata.almalinux.org",
					"tags": [
						"cve"
					],
					"url": "https://access.redhat.com/security/cve/CVE-2022-33987"
				}
			]
		}
	],
	"references": [
		{
			"name": "CVE-2022-32213",
			"source": "errata.almalinux.org",
			"tags": [
				"cve"
			],
			"url": "https://access.redhat.com/security/cve/CVE-2022-32213"
		},
		{
			"name": "2098556",
			"source": "errata.almalinux.org",
			"tags": [
				"bugzilla"
			],
			"url": "https://bugzilla.redhat.com/2098556"
		},
		{
			"name": "2102001",
			"source": "errata.almalinux.org",
			"tags": [
				"bugzilla"
			],
			"url": "https://bugzilla.redhat.com/2102001"
		},
		{
			"name": "ALSA-2022-6595",
			"source": "errata.almalinux.org",
			"tags": [
				"vendor-advisory"
			],
			"url": "https://errata.almalinux.org/9/ALSA-2022-6595.html"
		},
		{
			"name": "RHSA-2022:6595",
			"source": "errata.almalinux.org",
			"tags": [
				"third-party-advisory"
			],
			"url": "https://access.redhat.com/errata/RHSA-2022:6595"
		},
		{
			"name": "CVE-2020-28469",
			"source": "errata.almalinux.org",
			"tags": [
				"cve"
			],
			"url": "https://access.redhat.com/security/cve/CVE-2020-28469"
		},
		{
			"name": "CVE-2020-7788",
			"source": "errata.almalinux.org",
			"tags": [
				"cve"
			],
			"url": "https://access.redhat.com/security/cve/CVE-2020-7788"
		},
		{
			"name": "CVE-2022-29244",
			"source": "errata.almalinux.org",
			"tags": [
				"cve"
			],
			"url": "https://access.redhat.com/security/cve/CVE-2022-29244"
		},
		{
			"name": "1964461",
			"source": "errata.almalinux.org",
			"tags": [
				"bugzilla"
			],
			"url": "https://bugzilla.redhat.com/1964461"
		},
		{
			"name": "2105426",
			"source": "errata.almalinux.org",
			"tags": [
				"bugzilla"
			],
			"url": "https://bugzilla.redhat.com/2105426"
		},
		{
			"name": "2105428",
			"source": "errata.almalinux.org",
			"tags": [
				"bugzilla"
			],
			"url": "https://bugzilla.redhat.com/2105428"
		},
		{
			"name": "CVE-2022-32212",
			"source": "errata.almalinux.org",
			"tags": [
				"cve"
			],
			"url": "https://access.redhat.com/security/cve/CVE-2022-32212"
		},
		{
			"name": "CVE-2022-32214",
			"source": "errata.almalinux.org",
			"tags": [
				"cve"
			],
			"url": "https://access.redhat.com/security/cve/CVE-2022-32214"
		},
		{
			"name": "CVE-2022-32215",
			"source": "errata.almalinux.org",
			"tags": [
				"cve"
			],
			"url": "https://access.redhat.com/security/cve/CVE-2022-32215"
		},
		{
			"name": "1945459",
			"source": "errata.almalinux.org",
			"tags": [
				"bugzilla"
			],
			"url": "https://bugzilla.redhat.com/1945459"
		},
		{
			"name": "2007557",
			"source": "errata.almalinux.org",
			"tags": [
				"bugzilla"
			],
			"url": "https://bugzilla.redhat.com/2007557"
		},
		{
			"name": "2105422",
			"source": "errata.almalinux.org",
			"tags": [
				"bugzilla"
			],
			"url": "https://bugzilla.redhat.com/2105422"
		},
		{
			"name": "2105430",
			"source": "errata.almalinux.org",
			"tags": [
				"bugzilla"
			],
			"url": "https://bugzilla.redhat.com/2105430"
		},
		{
			"name": "CVE-2021-33502",
			"source": "errata.almalinux.org",
			"tags": [
				"cve"
			],
			"url": "https://access.redhat.com/security/cve/CVE-2021-33502"
		},
		{
			"name": "CVE-2021-3807",
			"source": "errata.almalinux.org",
			"tags": [
				"cve"
			],
			"url": "https://access.redhat.com/security/cve/CVE-2021-3807"
		},
		{
			"name": "CVE-2022-33987",
			"source": "errata.almalinux.org",
			"tags": [
				"cve"
			],
			"url": "https://access.redhat.com/security/cve/CVE-2022-33987"
		},
		{
			"name": "1907444",
			"source": "errata.almalinux.org",
			"tags": [
				"bugzilla"
			],
			"url": "https://bugzilla.redhat.com/1907444"
		}
	],
	"published": "2022-09-20T00:00:00Z",
	"modified": "2023-09-15T13:41:48Z",
	"data_source": "alma-osv"
}
